hedera-node/src/main/java/com/hedera/services/throttling/DeterministicThrottling.java

```diff
@@ -71,10 +76,18 @@ public boolean shouldThrottleQuery(HederaFunctionality queryFunction) {
 	public boolean shouldThrottleTxn(TxnAccessor accessor, Instant now) {
 		final var function = accessor.getFunction();
 		ThrottleReqsManager manager;
+
+		if (dynamicProperties.shouldThrottleByGas() && (function == ContractCreate || function == ContractCall)) {
```

Comment:
Given that the majority of the time, functions will not be `ContractCreate` or `ContractCall`, we can place that check as a first statement in the IF. That way we will skip the check of the dynamicProperties everytime unnecessarily, but check hat only if we are calling ContractCreate or Call